tutor/src/components/add-edit-question/terms-of-use.js

```diff
@@ -43,45 +45,63 @@ const StyledAddEditQuestionTermsOfUseModal = styled(AddEditQuestionTermsOfUseMod
     }
 `;
 
-const AddEditQuestionTermsOfUse = observer(({ ux }) => {
+const agreeTermsOfUse = () => {
+  const term = User.terms.get(TERMS_NAME);
+  if (term) {
+    User.terms.sign([term.id]);
+  }
+};
+
+const AddEditQuestionTermsOfUse = observer(({ show, onClose, displayOnly = false }) => {
   const [agree, setAgree] = useState(false);
+  const [termContent, setTermContent] = useState(null);
+
+  const term = useMemo(() => User.terms.get(TERMS_NAME),
+    [User.terms.get(TERMS_NAME).content]);
+  
   useEffect(() => {
     User.terms.fetch();
   }, []);
+  useEffect(() => {
+    setTermContent(term ? term.content : '');
+  }, [term.content]);
+  
   return (
     <StyledAddEditQuestionTermsOfUseModal
-      show={true}
+      show={show}
       backdrop="static"
-      onHide={() => ux.onDisplayModal(false)}
+      onHide={onClose}
       templateType="addEditQuestion">
       <Modal.Header closeButton>
         Terms of use
       </Modal.Header>
       <Modal.Body>
-
-        <div dangerouslySetInnerHTML={{ __html: ux.termsOfUse }} />
-
-        <CheckboxInput
-          className="i-agree"
-          onChange={() => setAgree(prevState => !prevState)}
-          label="I certify that these questions do not violate any copyright, trademark, or other intellectual property laws."
-          checked={agree}
-          standalone
-        />
-        <div className="buttons-wrapper">
-          <Button
-            variant="default"
-            className="cancel"
-            onClick={() => ux.onDisplayModal(false)}>
+        <div dangerouslySetInnerHTML={{ __html: termContent }} />
+        {!displayOnly && 
+        <>
+          <CheckboxInput
+            className="i-agree"
+            onChange={() => setAgree(prevState => !prevState)}
+            label="I certify that these questions do not violate any copyright, trademark, or other intellectual property laws."
+            checked={agree}
+            standalone
+          />
+          <div className="buttons-wrapper">
+            <Button
+              variant="default"
+              className="cancel"
+              onClick={onClose}>
                 Cancel
-          </Button>
-          <Button
-            variant="primary"
-            disabled={!agree}
-            onClick={ux.agreeTermsOfUse}>
-              Done
-          </Button>
-        </div>
+            </Button>
+            <Button
+              variant="primary"
+              disabled={!agree}
+              onClick={agreeTermsOfUse}>
+              Accept and continue
+            </Button>
+          </div>
+        </>
+        }
       </Modal.Body>
```

Comment:
instead of a huge if block, how about extracting to a TermAgreement component and render that if needed?